src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -0,0 +1,871 @@
+                public static void Main()
+                {
+                    var c = new C();
+                    c.InterceptableMethod("call site");
+                }
+            }
+
+            static class D
+            {
+                [InterceptsLocation("Program.cs", 14, 10)]
+                public static C Interceptor1(this C i1, string param) { Console.Write("interceptor " + param); return i1; }
+            }
+            """;
+        var verifier = CompileAndVerify(new[] { (source, "Program.cs"), s_attributesSource }, expectedOutput: "interceptor call site");
+        verifier.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void InterceptableStaticMethod_InterceptorInstanceMethod()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            static class Program
+            {
+                public static void Main()
+                {
+                    C.InterceptableMethod("call site");
+                }
+            }
+            
+            class C
+            {
+                [Interceptable]
+                public static void InterceptableMethod(string param) { Console.Write("interceptable " + param); }
+
+                [InterceptsLocation("Program.cs", 7, 10)]
+                public void Interceptor1(string param) { Console.Write("interceptor " + param); }
+            }
+            """;
+        var comp = CreateCompilation(new[] { (source, "Program.cs"), s_attributesSource });
+        comp.VerifyEmitDiagnostics(
+            // Program.cs(17,6): error CS27012: Interceptor must not have a 'this' parameter because 'C.InterceptableMethod(string)' does not have a 'this' parameter.
+            //     [InterceptsLocation("Program.cs", 7, 10)]
+            Diagnostic(ErrorCode.ERR_InterceptorMustNotHaveThisParameter, @"InterceptsLocation(""Program.cs"", 7, 10)").WithArguments("C.InterceptableMethod(string)").WithLocation(17, 6));
+    }
+
+    // PROTOTYPE(ic): test intercepting an extension method with a non-extension method. Perhaps should be an error for simplicity even if calling in non-reduced form.
+
+    [Fact]
+    public void ArgumentLabels()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            class C
+            {
+                [Interceptable]
+                public void InterceptableMethod(string s1, string s2) { Console.Write(s1 + s2); }
+            }
+
+            static class Program
+            {
+                public static void Main()
+                {
+                    var c = new C();
+                    c.InterceptableMethod(s2: "World", s1: "Hello ");
+                }
+            }
+
+            static class D
+            {
+                [InterceptsLocation("Program.cs", 14, 10)]
+                public static void Interceptor1(this C c, string s1, string s2) { Console.Write("interceptor " + s1 + s2); }
```

Comment:
>string s1, string s2 [](http://example.com/codeflow?start=58&length=20)

What if the names are reversed here? (Is there any requirement that parameter names of the methods match?)

Reply:
Added a note to test this.